Add delay between nginx reloads

alexsong93 opened the pull request:
Added logic in the subscribe function to reload nginx only if >1 second has passed since last nginx reload. Fixes #26. @mhamann @lostinthestory @DavidMGreen PTAL

lostinthestory commented:
@alexsong93 It seems that if the first time through, I fail the 1 second check, but then another API update doesn't come in for a long period of time, nginx would not reload to know about the first API for that long period of time. I would think we would almost need to implement it like you implemented the health check. On first request, set a flag that we need to reload and set a timer to reload in one second. On subsequent requests, see if we're already waiting to reload. If so, just update the file mgmt portion. Then just before we reload, set the flag back to false so the next API will trigger another reload in one second.

alexsong93 commented:
@lostinthestory I believe that's exactly what the code is doing - the `redisUpdated` flag is set to true inside the `else` block when redis is updated, and it is set back to false after the if/else block, if `redisUpdated == true and timeDiff >= 1`. The 1 second check is done regardless of whether or not redis has been updated.

I think the problem here is that I changed the timeout for `redisSubClient` to 10 seconds, from 1 second. This means that `redisSubClient:read_reply()` will block for 10 seconds if it doesn't see any updates. I changed the timeout back to 1 second - it should work as expected now.

lostinthestory commented:
Ah, I seem to have blocked the `redisUpdated` out of my vision altogether when I was looking this over.
